### ISSUE TITLE
httpd logs a warning about ignored failed future

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/seastar#250]
```
seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
```

It's when "joined" is destroyed here:

```
future<> process() {
            // Launch read and write "threads" simultaneously:
            return when_all(read(), respond()).then(
                    [] (std::tuple<future<>, future<>> joined) {
                        // FIXME: notify any exceptions in joined?
                        return make_ready_future<>();
                    });
        }
```

```
report_failed_future(std::__exception_ptr::exception_ptr) at /usr/src/debug/scylla-1.7.rc0/seastar/core/reactor.cc:3646
future<>::~future() at /usr/src/debug/scylla-1.7.rc0/seastar/core/future.hh:772
 (inlined by) std::_Head_base<1ul, future<>, false>::~_Head_base() at /opt/scylladb/include/c++/5.3.1/tuple:102
std::_Tuple_impl<1ul, future<> >::~_Tuple_impl() at /opt/scylladb/include/c++/5.3.1/tuple:339
 (inlined by) std::_Tuple_impl<0ul, future<>, future<> >::~_Tuple_impl() at /opt/scylladb/include/c++/5.3.1/tuple:180
 (inlined by) std::tuple<future<>, future<> >::~tuple() at /opt/scylladb/include/c++/5.3.1/tuple:596
 (inlined by) apply_helper<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<std::tuple<future<>, future<> > >&&, std::integer_sequence<unsigned long, 0ul> >::apply({lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >) at /usr/src/debug/scylla-1.7.rc0/seastar/core/apply.hh:34
 (inlined by) auto apply<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<future<>, future<> > >(httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >&&) at /usr/src/debug/scylla-1.7.rc0/seastar/core/apply.hh:42
 (inlined by) future<> futurize<future<> >::apply<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<future<>, future<> > >(httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >&&) at /usr/src/debug/scylla-1.7.rc0/seastar/core/future.hh:1267
 (inlined by) _ZZN6futureIJSt5tupleIJS_IJEES1_EEEE4thenIZN5httpd11http_server10connection7processEvEUlS2_E_S1_EET0_OT_ENUlSB_E_clI12future_stateIJS2_EEEEDaSB_ at /usr/src/debug/scylla-1.7.rc0/seastar/core/future.hh:872
 (inlined by) _ZN12continuationIZN6futureIJSt5tupleIJS0_IJEES2_EEEE4thenIZN5httpd11http_server10connection7processEvEUlS3_E_S2_EET0_OT_EUlSC_E_JS3_EE3runEv at /usr/src/debug/scylla-1.7.rc0/seastar/core/future.hh:390
reactor::run_tasks(circular_buffer<std::unique_ptr<task, std::default_delete<task> >, std::allocator<std::unique_ptr<task, std::default_delete<task> > > >&) at /usr/src/debug/scylla-1.7.rc0/seastar/core/reactor.cc:2023
reactor::run() at /usr/src/debug/scylla-1.7.rc0/seastar/core/reactor.cc:2465
smp::configure(boost::program_options::variables_map)::{lambda()#3}::operator()() const [clone .constprop.4079] at /usr/src/debug/scylla-1.7.rc0/seastar/core/reactor.cc:3394
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/5.3.1/functional:2271
 (inlined by) posix_thread::start_routine(void*) at /usr/src/debug/scylla-1.7.rc0/seastar/core/posix.cc:50
```

**Comment by slivne** [comment]
@amnonh please take a look

**Comment by tzach** [comment]
I'm able to reproduce the issue locally:

```
sudo docker run --name some-scylla -d scylladb/scylla:1.6.0 --memory 1G --smp 1
```

run monitoring from `scylla-grafana-monitoring` 

```
./start-all.sh 
```
(after updating `prometheus/scylla_servers.yml`)

run stress:
```
sudo docker exec -it some-scylla cassandra-stress write duration=3m -rate threads=500 -node 172.17.0.2
```

from the logs:
```
WARN  2017-02-26 09:23:45,447 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,464 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,475 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,493 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,503 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,521 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,549 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,557 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:23:45,559 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
WARN  2017-02-26 09:24:02,591 [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)
```

**Comment by slivne** [comment]
@amnonh can you give it a try - I suspect we may have someother bug causing this to happen - we need to find the source - not only handle the error.

**Comment by kongove** [comment]
@roydahan After disabled monitor, the seastar error didn't happen.

I also tried to use a big machine, but the error still occurred after job run less then 1 hour.

old: gce_instance_type_monitor: 'n1-standard-1'
new: gce_instance_type_monitor: 'n1-standard-2'

I checked in GCE page, the cpu is only used about 10% in the test.
![screen shot 2017-03-09 at 7 43 25 am](https://cloud.githubusercontent.com/assets/309708/23729349/59e10f68-049c-11e7-82a9-6185807282b0.png)

**Comment by nirmaayan** [comment]
Happen also for investing.com [ticket#111](https://supportscylladb.zendesk.com/agent/tickets/111)

**Comment by kongove** [comment]
I can still reproduce this bug in [longevity-gce test of 1.7.rc2-0.20170328.9b26a57 ](http://jenkins.cloudius-systems.com/job/scylla-1.7-longevity-gce/40/label=gce/) 

@slivne @roydahan 

The node isn't readded after Decommission tests, I'm studying if it's an environment issue(GCE).

```
2017-03-29 21:39:04,498 DEBUG| [10.240.0.35] [stdout] Mar 29 21:39:04 warning|  [shard 8] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe), backtrace: 0x4ec8f3, 0x4ec951, 0x57b93f, 0x82d7a9, 0x4e1d5f, 0x5318b2, 0x5330f0, 0x58e9fd, 0x7f089d323dc4, 0x7f089d05273c

[scylla-test@longevity-1-7-gce-amosbranch-scylla-db-node-8c6764d9-000 ~]$ addr2line -Cfpi -e /usr/lib/debug/usr/bin/scylla.debug 0x4ec8f3, 0x4ec951, 0x57b93f, 0x82d7a9, 0x4e1d5f, 0x5318b2, 0x5330f0, 0x58e9fd, 0x7f089d323dc4, 0x7f089d05273c
current_backtrace() at /usr/src/debug/scylla-1.7.rc2/seastar/core/reactor.cc:3623
report_failed_future(std::__exception_ptr::exception_ptr) at /usr/src/debug/scylla-1.7.rc2/seastar/core/reactor.cc:3648
future<>::~future() at /usr/src/debug/scylla-1.7.rc2/seastar/core/future.hh:772
 (inlined by) std::_Head_base<1ul, future<>, false>::~_Head_base() at /opt/scylladb/include/c++/5.3.1/tuple:102
std::_Tuple_impl<1ul, future<> >::~_Tuple_impl() at /opt/scylladb/include/c++/5.3.1/tuple:339
 (inlined by) std::_Tuple_impl<0ul, future<>, future<> >::~_Tuple_impl() at /opt/scylladb/include/c++/5.3.1/tuple:180
 (inlined by) std::tuple<future<>, future<> >::~tuple() at /opt/scylladb/include/c++/5.3.1/tuple:596
 (inlined by) apply_helper<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<std::tuple<future<>, future<> > >&&, std::integer_sequence<unsigned long, 0ul> >::apply({lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >) at /usr/src/debug/scylla-1.7.rc2/seastar/core/apply.hh:34
 (inlined by) auto apply<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<future<>, future<> > >(httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >&&) at /usr/src/debug/scylla-1.7.rc2/seastar/core/apply.hh:42
 (inlined by) future<> futurize<future<> >::apply<httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}, std::tuple<future<>, future<> > >(httpd::http_server::connection::process()::{lambda(std::tuple<future<>, future<> >)#1}&&, std::tuple<std::tuple<future<>, future<> > >&&) at /usr/src/debug/scylla-1.7.rc2/seastar/core/future.hh:1267
 (inlined by) _ZZN6futureIJSt5tupleIJS_IJEES1_EEEE4thenIZN5httpd11http_server10connection7processEvEUlS2_E_S1_EET0_OT_ENUlSB_E_clI12future_stateIJS2_EEEEDaSB_ at /usr/src/debug/scylla-1.7.rc2/seastar/core/future.hh:872
 (inlined by) _ZN12continuationIZN6futureIJSt5tupleIJS0_IJEES2_EEEE4thenIZN5httpd11http_server10connection7processEvEUlS3_E_S2_EET0_OT_EUlSC_E_JS3_EE3runEv at /usr/src/debug/scylla-1.7.rc2/seastar/core/future.hh:390
reactor::run_tasks(circular_buffer<std::unique_ptr<task, std::default_delete<task> >, std::allocator<std::unique_ptr<task, std::default_delete<task> > > >&) at /usr/src/debug/scylla-1.7.rc2/seastar/core/reactor.cc:2025
reactor::run() at /usr/src/debug/scylla-1.7.rc2/seastar/core/reactor.cc:2467
smp::configure(boost::program_options::variables_map)::{lambda()#3}::operator()() const [clone .constprop.4079] at /usr/src/debug/scylla-1.7.rc2/seastar/core/reactor.cc:3396
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/5.3.1/functional:2271
 (inlined by) posix_thread::start_routine(void*) at /usr/src/debug/scylla-1.7.rc2/seastar/core/posix.cc:50
?? ??:0
```

**Comment by penberg** [comment]
Is this issue resolved by commit d468ea5c627aa210b41cec522b3bc12c7cc0c3d2?

**Comment by amnonh** [comment]
I think it's closed
